crates/fuel-core/src/schema/tx/types.rs

```diff
@@ -212,6 +215,42 @@ impl SuccessStatus {
     }
 }
 
+#[derive(Debug)]
+pub struct SuccessDuringBlockProductionStatus {
+    pub tx_pointer: TxPointer,
+    pub tx_id: Option<TxId>,
```

Comment:
`tx_id` always will exist in pre-confirmation

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2752/commits/87aaf7ca6dc40434d97f12f90bf717edf891417e